pkg/gomod/selector.go

```diff
@@ -0,0 +1,104 @@
+package gomod
+
+import (
+	"log"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+	"knative.dev/test-infra/pkg/gowork"
+)
+
+var (
+	// ErrNoDomain is returned when no domain is provided.
+	ErrNoDomain = errors.New("no domain provided")
+)
+
+// Matcher matches a module name.
+type Matcher interface {
+	// Match returns true if the module name matches.
+	Match(modname string) bool
+}
+
+// MatcherFunc is a function that implements the Matcher interface.
+func MatcherFunc(fn matcherFunc) Matcher {
+	return fn
+}
+
+type matcherFunc func(modname string) bool
+
+func (fn matcherFunc) Match(modname string) bool {
+	return fn(modname)
+}
+
+// Selector is a set of matchers that can be used to select a subset of modules.
```

Comment:
With a list of Includes and Excludes, it feels worth documenting whether "Include wins" or "Exclude wins", which also controls whether the default behavior is to include or exclude.

Reply:
Agree. A bit more docs, could be beneficial.